### PR TITLE
Add documentation for Batch to README and add commands to fill gaps

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#2584 

## Purpose/Implementation Notes

This updates the README with documentation for the new changes caused by switching to AWS Batch. I also noticed that a few sections didn't have an analogous command because they had relied on directly dispatching jobs to Nomad. I fixed this by adding one management command and tweaking another.

This also adds a timeout to the invocation of `download_dataset()` in the end to end tests because the API client PR landed.

## Types of changes

- Documentation
- New feature (non-breaking change which adds functionality)

## Functional tests

The tests pass, I've added tests, and I've even rerun end to end tests against a dev stack to make sure the download_dataset() change worked.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -667,21 +677,56 @@ terraform taint <your-entity-from-state-list>
...
+  In refine.bio each Job Queue uses a single compute environment, so if two jobs are placed in the same job queue they will be run in the same Compute Environment.
+* Job Definitions:
+  These are what specify the configuration to be used for each job type including what Docker Image will be used, what environment variables will be passed to it, what secrets it can access, and how many vCPUs and RAM it requires.
+
+refine.bio uses three types of job queues:
+* **Compendia Job Queue**:
+  This job queue is for running very large compendia-building jobs that require a large instance. The Compute Environment assigned to this queue is configured to provision very large instances.
+* **Smasher Job Queue**:
+  This job queue is used for running smashing jobs.
+  Having a dedicated queue for smasher jobs is useful because it ensure they won't be blocked by processing jobs and the instance provisioned by its compute environment has enough resources to run of these jobs at a time and no more.
```
Comment (wvauclain) [inline README.md:700]: Run how many?

File: README.md [hunk 2]
```diff
@@ -339,49 +337,49 @@ Surveyor Jobs discover samples to download/process along with
...
+./foreman/run_management_command.sh survey_all -h
 ```
 
 The Surveyor can accept a single accession code from any of the source
 data repositories (e.g., Sequencing Read Archive,
  ArrayExpress, Gene Expression Omnibus):
 
 ```bash
-./foreman/run_surveyor.sh survey_all --accession <ACCESSION_CODE>
+./foreman/run_management_command.sh survey_all --accession <ACCESSION_CODE>
```
Comment (wvauclain) [inline README.md:356]: What happens when you survey something locally now that we ripped out Nomad? Might be good to touch on this briefly somewhere in the docs

File: README.md [hunk 3]
```diff
@@ -512,16 +510,16 @@ Or for more information run:
...
-nomad job dispatch -meta ORGANISM=DANIO_RERIO CREATE_QN_TARGET
+./run_management_command dispatch_qn_jobs --organisms=DANIO_RERIO,HOMO_SAPIENS
 ```
 
-To create QN targets for all organisms, do so with the dispatcher:
+To create QN targets for all organisms with enough processed samples:
 
 ```bash
-nomad job dispatch QN_DISPATCHER
+./run_management_command dispatch_qn_jobs
```
Comment (wvauclain) [inline README.md:522]:
```suggestion
./run_management_command.sh dispatch_qn_jobs
```

File: README.md [hunk 4]
```diff
@@ -512,16 +510,16 @@ Or for more information run:
 
 ### Creating Quantile Normalization Reference Targets
 
-If you want to quantile normalize combined outputs, you'll first need to create a reference target for a given organism. This can be done in a production environment with the following:
+If you want to quantile normalize combined outputs, you'll first need to create a reference target for a given organism or organisms. This can be done in a production environment by running the following on the Foreman instance:
 
 ```bash
-nomad job dispatch -meta ORGANISM=DANIO_RERIO CREATE_QN_TARGET
+./run_management_command dispatch_qn_jobs --organisms=DANIO_RERIO,HOMO_SAPIENS
```
Comment (wvauclain) [inline README.md:516]:
```suggestion
./run_management_command.sh dispatch_qn_jobs --organisms=DANIO_RERIO,HOMO_SAPIENS
```

File: foreman/data_refinery_foreman/foreman/management/commands/test_run_tximport.py [hunk 5]
```diff
@@ -9,34 +9,34 @@
...
 ) -> Dict:
-    """Create an experiment and associated objects and run tximport on it.
+    """Create an experiment and associated objects that tximport needs to run on it.
 
     Creates a sample for each accession contained in either input
     list. The samples in complete_accessions will be simlulated as
     already having salmon quant run on them. The samples in
     incomplete_accessions won't.
+
+    If the accession_codes parameter is supplied the
```
Comment (wvauclain) [inline foreman/data_refinery_foreman/foreman/management/commands/test_run_tximport.py:37]: Is this comment up-to-date?

**Review by wvauclain — CHANGES_REQUESTED** [review]
A couple of small questions, but otherwise it looks good to me

File: foreman/data_refinery_foreman/foreman/management/commands/test_run_tximport.py (hunk 5 above)
Comment (kurtwheeler) [reply foreman/data_refinery_foreman/foreman/management/commands/test_run_tximport.py:37]: Nope! I was going to do it like this but then I just ended up calling the different functions in the tests directly!

File: README.md (hunk 2 above)
Comment (kurtwheeler) [reply README.md:356]: This is a good question but I don't actually have a good answer. I just made https://github.com/AlexsLemonade/refinebio/issues/2775

Do you think I should address that before merging to dev?

Comment (wvauclain) [reply README.md:356]: I don't think it's worth blocking everything else, so we can probably put it off until next sprint

**Review by wvauclain — APPROVED** [review]
:rocket:

**Pull request merged**